File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--creating-shortcut-menus-for-bookmarks_4.vb
    Public Sub ButtonClick(ByVal control As Office.IRibbonControl)
        If control.Id = "BoldButton" Then
            If Globals.ThisDocument.selectedBookmark.Bold = _
                Globals.ThisDocument.WordTrue Then
                Globals.ThisDocument.selectedBookmark.Bold = _
                    Globals.ThisDocument.WordFalse
            Else
                Globals.ThisDocument.selectedBookmark.Bold = _
                    Globals.ThisDocument.WordTrue
            End If
        ElseIf control.Id = "ItalicButton" Then
            If Globals.ThisDocument.selectedBookmark.Italic = _
                Globals.ThisDocument.WordTrue Then
                Globals.ThisDocument.selectedBookmark.Italic = _
                    Globals.ThisDocument.WordFalse
            Else
                Globals.ThisDocument.selectedBookmark.Italic = _
                    Globals.ThisDocument.WordTrue
            End If
        End If
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--creating-shortcut-menus-for-bookmarks_4.cs
        public void ButtonClick(Office.IRibbonControl control)
        {
            if (control.Id == "BoldButton")
            {
                if (Globals.ThisDocument.selectedBookmark.Bold == 
                    Globals.ThisDocument.WordTrue)
                {
                    Globals.ThisDocument.selectedBookmark.Bold = 
                        Globals.ThisDocument.WordFalse;
                }
                else
                {
                    Globals.ThisDocument.selectedBookmark.Bold = 
                        Globals.ThisDocument.WordTrue;
                }
            }
            else if (control.Id == "ItalicButton")
            {
                if (Globals.ThisDocument.selectedBookmark.Italic == 
                    Globals.ThisDocument.WordTrue)
                {
                    Globals.ThisDocument.selectedBookmark.Italic = 
                        Globals.ThisDocument.WordFalse;
                }
                else
                {
                    Globals.ThisDocument.selectedBookmark.Italic = 
                        Globals.ThisDocument.WordTrue;
                }
            }

        }